fill/fill_station.cc

```diff
@@ -126,7 +126,24 @@ class TelemeterServiceImpl final : public FillStationTelemeter::Service
                        ServerWriter<FillStationTelemetry> *writer) override
     {
         spdlog::info("Received initial connection point for the fill-station telemetry service.\n");
-        while (true) {
+
+        std::chrono::steady_clock::time_point errorStartTime; // gives fill station 10 seconds to reconnect to the ground server 
+        bool errorTimerStarted = false;
+        constexpr std::chrono::seconds errorThreshold(10);
+        while (true) {  
+            if (context->IsCancelled()){ 
```

Comment:
Pretty sure once this is cancelled it will never be not cancelled again. When they reconnect, it will open a new stream with a new context that's entirely separate from the old one, so this would close the ball valve even if they reconnected